utils/service_discovery/sd_docker_backend.py

```diff
@@ -317,15 +348,39 @@ def get_tags(self, state, c_id):
                 tags.append('swarm_service:%s' % swarm_svc)
 
         elif Platform.is_rancher():
-            service_name = c_inspect.get('Config', {}).get('Labels', {}).get(RANCHER_SVC_NAME)
-            stack_name = c_inspect.get('Config', {}).get('Labels', {}).get(RANCHER_STACK_NAME)
-            container_name = c_inspect.get('Config', {}).get('Labels', {}).get(RANCHER_CONTAINER_NAME)
+            c_inspect = state.inspect_container(c_id)
+
+            container_name = c_inspect.get('Config', {}).get('Labels', {}).get(RancherUtil.CONTAINER_NAME_LABEL)
+
+            container_metadata = self.rancherutil.get_container_metadata(container_name=container_name)
+
+            service_name = c_inspect.get('Config', {}).get('Labels', {}).get(RancherUtil.SERVICE_NAME_LABEL) \
+                or container_metadata.get('service_name')
+
+            stack_name = c_inspect.get('Config', {}).get('Labels', {}).get(RancherUtil.STACK_NAME_LABEL) \
+                or container_metadata.get('stack_name')
+
+            if container_name:
+                tags.append('rancher_container:%s' % container_name)
             if service_name:
                 tags.append('rancher_service:%s' % service_name)
             if stack_name:
                 tags.append('rancher_stack:%s' % stack_name)
-            if container_name:
-                tags.append('rancher_container:%s' % container_name)
+
+            host_labels = self.rancherutil.get_labels_for_host()
+
+            for (k, v) in host_labels.iteritems():
+                if k == RancherUtil.HOST_AGENT_IMAGE_LABEL:
+                    tags.append('rancher_host_agent_image:%s' % v)
+
+                elif k == RancherUtil.HOST_DOCKER_VERSION_LABEL:
+                    tags.append('rancher_host_docker_version:%s' % v)
+
+                elif k == RancherUtil.HOST_LINUX_KERNEL_VERSION_LABEL:
+                    tags.append('rancher_host_linux_kernel_version:%s' % v)
```

Comment:
I don't think those 3 fit as tags in service discovery. They're good as host metadata but could you remove them from here?